File: Algorithms/Sorting Algorithms/Python/mergeSort.py
def mergeSort(array):
    if(len(array)>1):       #only proceed if array size is greater than 1
        
        m=len(array)//2
        left=array[:m]
        right=array[m:]
        mergeSort(left)
        mergeSort(right)

        i=j=k=0
        while(i<len(left) and j<len(right)):  #go through the 2 arrays
            if(left[i]<right[j]): 
                array[k] = left[i] 
                i+=1
            else: 
                array[k] = right[j] 
                j+=1
            k+=1
    
        while(i<len(left)):     #if left array has left over elements
            array[k]=left[i]
            i+=1
            k+=1

        while(j<len(right)):    #if right array has left over elements
            array[k]=right[j]
            j+=1
            k+=1

a=[1,2,4,3,0,6,56,7,8,12,11] #edit as needed
mergeSort(a)
print(a)

File: Data Structures/Scala/merge_sort.scala
def mergeSort(arr: List[Int]): List[Int] = {
  def merge(left: List[Int], right: List[Int]): List[Int] = (left, right) match {
    case (Nil, _) => right
    case (_, Nil) => left
    case (lh :: lt, rh :: rt) =>
      if (lh < rh) lh :: merge(lt, right)
      else rh :: merge(left, rt)
  }

  val n = arr.length / 2
  if (n == 0) arr
  else {
    val (left, right) = arr.splitAt(n)
    merge(mergeSort(left), mergeSort(right))
  }
}
